### PR TITLE
Fix survey

### PR DESCRIPTION
**Pull request by ccomb**
backport of #6243

### CODE REVIEW

**Comment by gurneyalex** [comment]
@ccomb could you please rebase?

**Comment by ccomb** [comment]
@gurneyalex I just rebased and push force. Hope it's ok?

**Comment by yvaucher** [comment]
@ccomb Yep this is just what we mean by rebasing

**Comment by yvaucher** [comment]
:+1:
By the way translation for this string is not present in pot file, I wonder if an other fix should be done for it.

File: addons/survey/wizard/survey_send_invitation.py [hunk 1]
```diff
@@ -71,20 +71,21 @@ def default_get(self, cr, uid, fields_list, context=None):
             survey_id = sur.id
         if msg:
             raise osv.except_osv(_('Warning!'), _('The following surveys are not in open state: %s') % msg)
-        data['mail'] = _('''
-Hello %%(name)s, \n\n
-Would you please spent some of your time to fill-in our survey: \n%s\n
+        data['mail'] = _('''Hello %%(name)s,
+
+Would you please spend some of your time to fill-in our survey: %(survey)s
```
Comment (yvaucher) [inline addons/survey/wizard/survey_send_invitation.py]: An update in POT file is required

Comment (ccomb) [inline addons/survey/wizard/survey_send_invitation.py]: @yvaucher update in POT done!

**Comment by yvaucher** [comment]
:+1:

**Comment by gurneyalex** [comment]
@ccomb actually, it's not only the POT file which needs changing: the msgid string needs updating in all the PO files too, and likely the translations need updating too. 

As is now, I'm concerned that trying to send a survey for users with a language != english will cause a crash (error in the % operator) or an untranslated string.

**Comment by yvaucher** [comment]
@gurneyalex you are right I didn't think about issue with change on % operator thus I remove my approval

**Comment by ccomb** [comment]
After discussing IRL with @gurneyalex and @StefanRijnhart I'm still unsure about the correct way to update po files. Normally I should just msgmerge everything with something like `for i in *po; do msgmerge -U $i survey.pot; done`, but in this case all the po files will be reformated and the diff will be larger than needed, probably causing further merge conflicts.  Maybe transifex will change things on this topic?
@odony according to you what is the recommended way to do this?

**Comment by ccomb** [comment]
@StefanRijnhart could you review the latests commits to see if that's what you expected?
I created a l10n_OCB module with just the modified translation and reverted the previous POT file as it was previously

**Comment by odony** [comment]
As noted on upstream #6243: I think you should avoid doing any change of interpolation specifiers in translated terms when you write a fix. This would discard all existing translations with no good reason. If you use the same specifiers, you can perform a simple search&replace of the msgid in the POT **and** all PO files, and that will be correct most of the time (especially when fixing typos - translators don't usually replicate typos when they translate)
You cannot/shouldn't use `msgmerge` to do this, `sed -i` should do the job. Watch out for small strings in those cases, there's a small risk of introducing duplicate msgids - not here though.
You can always validate modified files with `msgfmt -c`.

File: addons/l10n_OCB/__init__.py [hunk 2]
```diff
@@ -0,0 +1 @@
+#
```
Comment (StefanRijnhart) [inline addons/l10n_OCB/__init__.py:1]: Can be completely empty, actually.

File: addons/l10n_OCB/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    'depends': [
+        'base',
+    ],
+    'data': [],
+    'images': [],
+    'update_xml': [],
+    'demo': [],
+    'installable': True,
+    'auto_install': True,
+    'application': True,
```
Comment (StefanRijnhart) [inline addons/l10n_OCB/__openerp__.py]: Not an application, really.

File: addons/survey/wizard/survey_send_invitation.py (hunk 1 above)
Comment (StefanRijnhart) [inline addons/survey/wizard/survey_send_invitation.py]: no dash in 'fill-in', just 'to fill in our survey:'

**Comment by StefanRijnhart** [comment]
Some nags, but otherwise good. Let's not merge this until we have the Transifex for the new translation module set up.

**Comment by gurneyalex** [comment]
runbot chokes on the following warning

2015-06-05 12:03:18,270 26786 WARNING 2948314-197-0d5f8a-all openerp.addons.base.module.module: module l10n_OCB: description is empty !

File: addons/l10n_OCB/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+##############################################################################
+{
+    'name': 'OCB translations',
+    'version': '0.1',
+    'author': 'OCA',
+    'sequence': 200,
+    'category': 'Localization',
+    'website': 'http://odoo-community.org',
+    'summary': 'Module used to host translations specific to OCB.',
+    'description': "",
```
Comment (gurneyalex) [inline addons/l10n_OCB/__openerp__.py:29]: putting a description here, or replacing this key with a README.rst file should get runbot green.

**Comment by pedrobaeza** [comment]
I would call it the module _i18n_ocb_, not _l10n_ocb_

**Comment by StefanRijnhart** [comment]
@pedrobaeza you are right, I was confused. i18n_ocb it should be.

**Pull request closed without merging**